sandpack-client/src/clients/static/index.ts

```diff
@@ -81,17 +81,22 @@ export class SandpackStatic extends SandpackClient {
     }
   }
 
-  private injectDocType(content: FileContent): FileContent {
+  private validateHtml(content: FileContent): FileContent {
     // Make it a string
     let contentString = readBuffer(content);
 
-    // Add the DOCTYPE tag
-    const docTypeRegex = /<!DOCTYPE*>/gi;
-    if (!docTypeRegex.test(contentString)) {
-      contentString = `<!DOCTYPE html>\n${content}`;
-    }
+    // Check how long DOMParser takes
+    console.time('domparser');
+
+    const domParser = new DOMParser();
+    const doc = domParser.parseFromString(contentString, "text/html");
+    doc.documentElement.setAttribute("lang", "en");
+
+    const html = doc.documentElement.outerHTML;
+
+    console.timeEnd('domparser');
 
-    return contentString;
+    return `<!DOCTYPE html>\n${html}`;
```

Comment:
The `documentElement.outerHTML` property is a string that includes everything within the `<html>` tags, inclusively. But the `<!DOCTYPE>` sits outside the HTML tag. Therefore, we need to add it back in, even if the user included it in their content.